### PR TITLE
Update Segment Page call traits

### PR DESCRIPTION
**Pull request by TimJones97**
## Motivation
The Segment Page call tracks standard page views and meta data related to apps. It's currently deployed in testnet, but needs to be brought into production on launch day. 

The existing solution needs to be enhanced to 

1. Pick up [common properties](https://segment.com/docs/connections/spec/common/) in all page views
2. Create new properties to reflect the state of the page.

https://mycelium-mex.atlassian.net/browse/SWAP-149

## Changelog

Add the following traits to the Page calls that occur on Trade/Buy TLP/Rewards pages

### Trade

- Graph Time: 5m, 15m,1h, 4h, 1d
- Table View: Positions, Orders, Trades
- Market: BTC / USD, ETH / USD
- Market Position: Long, Short, Swap
- Pay Currency: BTC, ETH, WETH, etc
- Pay Currency Modal
- Receive Currency: BTC, ETH, etc
- Receive Currency Modal
- Leverage Slider: Numeric

### Buy TLP

- Action: Buy, Sell
- Pay Currency: BTC, ETH, WETH, etc
- Receive Currency: BTC, ETH, etc
- Buy with: BTC, ETH, etc. 

### Rewards

- Week: Numeric

### CODE REVIEW

File: src/components/Glp/GlpSwap.js [hunk 1]
```diff
@@ -649,6 +652,33 @@ export default function GlpSwap(props) {
     }
   };
 
+  const [pageTracked, setPageTracked] = useState(false);
+
+  const dataElements = [chainId, isBuying, pageTracked, swapTokenAddress, history.location.hash];
+  const elementsLoaded = Object.values(dataElements).every((element) => element !== undefined);
```
Comment (dospore) [inline src/components/Glp/GlpSwap.js:658]: JS freaky, could you also just use dataElements.every?

Comment (TimJones97) [reply src/components/Glp/GlpSwap.js:655]: cannot because that always returns true, i need to test for the values of the json objects

Comment (TimJones97) [reply src/components/Glp/GlpSwap.js:655]: e.g. 
```
{
  chainId: 421611,
  tableView: undefined,
}
```
^ still returns true

Comment (dospore) [reply src/components/Glp/GlpSwap.js:655]: but if dataElements is an array the values are just the items in the array?

**Review by dospore — APPROVED** [review]
LGTM

**Review by dospore — APPROVED** [review]

**Pull request merged**